### DISCUSSION TITLE
Handling components state externally

### DISCUSSION THREAD

**Discussion by kleinpetr** [primefaces/community#3171] [PrimeVue]
Hello I am wondering how to interact with components, for example Speed Dial, externally, out of slots, the typical usage in vue is using `exposes` at the component level which can be accessed from the parent something like 

```
const myDial = ref()

function openMenu() {
  myDial.toggle()
} 

...

<SpeedDial ref="myDial" ...  />
<button @click="openMenu()">Open</button>
```

I would need to toggle visibility of this component and also others, such as menus, etc.. Thanks for clarification

**Comment by j0rgedev** [comment]
Hello @kleinpetr! I wonder, why don’t you try using the [component emitters](https://primevue.org/speeddial/#api.speeddial.emits)? For example, SpeedDial provides show and hide events that you can use to toggle its visibility without directly accessing the component via ref

**Reply by kleinpetr** [reply]
Not sure if we both understand 'emits' correctly but I guess emits should be passed as callbacks from the component when the parent can listen and react to them not firing internal functions, for that I think there is `expose` - which expose component methods they can be called from the parent.

As you can see in the docs 
- https://vuejs.org/guide/components/events
- https://vuejs.org/api/sfc-script-setup.html#defineexpose
